Minor bugfix

ExplicitChimp opened the pull request:
8/10/23 version 1.7.2: ExplicitChimp: Forked repos, Added #include <avr/dtostrf.h> @ line 43 and corrected lines 225 to 228 (inclusive) to read "readBit(RLY_I2C_ADD, ...)"   <----This was incorrect implementation!!!

I made a mistake in this commit so update to version 1.7.3:

28/10/23 version 1.7.3: ExplicitChimp: Forked repos, Added #include <avr/dtostrf.h> @ line 43 and corrected lines 225 to 228 (inclusive) to read "readBit(add, ...)"  <---- This is now the corrected implementation!!!

ExplicitChimp committed:
Update mLink.h

ExplicitChimp committed:
Update mLink.h

ExplicitChimp committed:
Update mLink.h

Corrected an error made when updating to version 1.7.2  (See version info)

HobbyComponents commented:
Thanks for spotting this mistake. Can I first ask if there is a reason why you have uncommented line 42 (#include <avr/dtostrf.h>) ?

ExplicitChimp commented:
Good Afternoon,

No problem.  The reason I added (#include <avr/dtostrf.h>) is because I am
using a an Arduino Nano 33 IOT, and I have to include this for any code
that calls (dtostrf()).  If I don't then it flags up a load of errors when
compiling.

If I comment out that line in mLink.h, this is what is outputted in the
terminal, when compiling:

"Compilation error: 'dtostrf' was not declared in this scope"

I believe the function dtostrf() was written for AVR architectures, but now
that things have move on, the work around is to include the function, when
using libraries that utilise said function.

I hope this makes sense.

Kind regards,

Wes (ExplicitChimp)

On Wed, 17 Jan 2024, 14:56 Hobby Components, ***@***.***>
wrote:

> Thanks for spotting this mistake. Can I first ask if there is a reason why
> you have uncommented line 42 (#include <avr/dtostrf.h>) ?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/HobbyComponents/mLink/pull/2#issuecomment-1895981982>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AS5KATAHNHLD2MRY5I74L23YO7RCFAVCNFSM6AAAAABB5NKPBOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQOJVHE4DCOJYGI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

HobbyComponents commented:
Thanks again. Both the bug fix and the dtostrf include changes should be added but the problem is if implemented as suggested the include fix will cause a compile error for avr boards. 

I'll have a think of the best way to include it for non-avr boards and get back to you.

ExplicitChimp commented:
Always happy to help.

For the dtostrf issue I have changed my code to:
#ifdef __AVR__
//Do nothing
#else
#include <avr/dtostrf.h>
#endif

This works for me and has zero compile errors.  I cannot test it on an AVR
board, but I believe it will work.



On Wed, 17 Jan 2024 at 18:11, Hobby Components ***@***.***>
wrote:

> Thanks again. Both the bug fix and the dtostrf include changes should be
> added but the problem is if implemented as suggested the include fix will
> cause a compile error for avr boards.
>
> I'll have a think of the best way to include it for non-avr boards and get
> back to you.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/HobbyComponents/mLink/pull/2#issuecomment-1896343572>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AS5KATDPSXL2ZBRFDREVIMDYPAH4BAVCNFSM6AAAAABB5NKPBOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQOJWGM2DGNJXGI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

ExplicitChimp committed:
Update mLink.h

Added #ifdef statement to capture requirements for non AVR boards to include <avr/dtostrf.h>

HobbyComponents commented:
Excellent. I've just done a test on a Uno and seems to be working fine. Thanks for your help!
